File: src/main/java/com/dh/ondot/schedule/core/exception/NotFoundScheduleException.java
package com.dh.ondot.schedule.core.exception;

import com.dh.ondot.core.exception.NotFoundException;

import static com.dh.ondot.core.exception.ErrorCode.NOT_FOUND_SCHEDULE;

public class NotFoundScheduleException extends NotFoundException {
    public NotFoundScheduleException(Long scheduleId) {
        super(NOT_FOUND_SCHEDULE.getMessage().formatted(scheduleId));
    }

    @Override
    public String getErrorCode() {
        return NOT_FOUND_SCHEDULE.name();
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/core/exception/NotFoundScheduleException.kt
package com.dh.ondot.schedule.core.exception

import com.dh.ondot.core.exception.ErrorCode.NOT_FOUND_SCHEDULE
import com.dh.ondot.core.exception.NotFoundException

class NotFoundScheduleException(scheduleId: Long) :
    NotFoundException(NOT_FOUND_SCHEDULE.message.format(scheduleId)) {
    override val errorCode: String get() = NOT_FOUND_SCHEDULE.name
}
